test/index.ts

```diff
@@ -135,7 +137,45 @@ it('block to storage', async () => {
 	const storedValue = Bytes.fromByteArray(rlpDecode(storedValueRlp) as Uint8Array)
 	const expectedValue = Bytes.fromByteArray([...Bytes.fromUnsignedInteger(11, 32), ...Bytes.fromUnsignedInteger(7, 112), ...Bytes.fromUnsignedInteger(5, 112)]).toUnsignedBigint()
 	expect(storedValue.toUnsignedBigint()).toEqual(expectedValue)
-}, 999999)
+})
+
+it('UniswapOracle on-chain hashes', async () => {
+	// setup
+	const rpc = await createMemoryRpc(jsonRpcEndpoint, gasPrice)
+	const {uniswapContract} = await createUniswap(rpc)
+
+	// get the block
+	const block = await rpc.getBlockByNumber(false, 'latest')
+	if (block === null) throw new Error(`received null latest block from node`)
+	// execute a transaction on-chain to force a block minted after the one we just fetched, so blockhash(block.number) works.
+	await new Promise(resolve => setTimeout(resolve, 1000))
+	await rpc.sendEth(await rpc.addressProvider(), 0n)
+	const rlpBlock = rlpEncodeBlock(block)
+
+	const uniswapContractAddressHash = await keccak256.hash(Bytes.fromUnsignedInteger(uniswapContract.address, 160))
+
+	const uniswapOracleContractAddress = await deploy(rpc, 'UniswapOracle.sol', 'UniswapOracle', ["address", "uint256"], [uniswapContract.address, BigInt(Date.now()) ])
+	const uniswapOracleContract = new UniswapOracle(new DependenciesImpl(rpc), uniswapOracleContractAddress)
+
+	// check hash created in constructor
+	const storedUniswapContractAddressHash = await uniswapOracleContract.uniswapV2PairHash_()
+	expect(storedUniswapContractAddressHash).toEqual(uniswapContractAddressHash)
+
+	// check constant hashes, generate proofs
+	const reserveTimestampSlotProof = await rpc.getProof(uniswapContract.address, [8n], block.number!)
+	const accountNodesRlp = rlpEncode(reserveTimestampSlotProof.accountProof.map(rlpDecode))
+	const constantReserveTimestampSlotHash = await uniswapOracleContract.reserveTimestampSlotHash_()
+	const reserveTimestampSlotStorageNodesRlp = rlpEncode(reserveTimestampSlotProof.storageProof[0].proof.map(x => rlpDecode(x)))
+	expect(constantReserveTimestampSlotHash).toEqual(await keccak256.hash(Bytes.fromUnsignedInteger(8n, 256)))
+
+	const price0SlotProof = await rpc.getProof(uniswapContract.address, [9n], block.number!)
+	const constantPrice0SlotHash = await uniswapOracleContract.price0SlotHash_()
+	const price0SlotStorageNodesRlp = rlpEncode(price0SlotProof.storageProof[0].proof.map(x => rlpDecode(x)))
+	expect(constantPrice0SlotHash).toEqual(await keccak256.hash(Bytes.fromUnsignedInteger(9n, 256)))
+
+	const price = await uniswapOracleContract.getPrice_(rlpBlock, accountNodesRlp, reserveTimestampSlotStorageNodesRlp, price0SlotStorageNodesRlp)
+	console.log({price})
```

Comment:
For later PR: Change this to an expect rather than a log.  Definitely not a blocker.

Reply:
the price isn't correct, that's why i log.